server.go

```diff
@@ -673,44 +669,108 @@ func (s *Server) applyCreateShardIfNotExists(m *messaging.Message) (err error) {
 		return ErrRetentionPolicyNotFound
 	}
 
-	// If we can match to an existing shard date range then just ignore request.
-	for _, sh := range rp.Shards {
-		if timeBetweenInclusive(c.Timestamp, sh.StartTime, sh.EndTime) {
-			return nil
-		}
+	// If we can match to an existing shard group date range then just ignore request.
```

Comment:
This can happen because two unapplied "create shard" commands were in the log, right? It can happen if multiple requests hit the cluster around the same time, right?